### PR TITLE
Refresh memory management docs

### PR DESCRIPTION
**Pull request by abrestic-rivos**
This commit updates `MEMORY.md` with a more current overview of the various memory management flows and the overall page state machine.

### CODE REVIEW

File: MEMORY.md [hunk 1]
```diff
@@ -42,132 +65,279 @@ classDiagram
...
+        H->>S: TvmAddSharedMemoryRegion
+        S->>S: Mark region in guest address space as shared
+    end
+    loop For each vCPU
+        H->>S: TvmAddCpu
+    end
+    H->>S: Finalize
+```
+
+After VM finaliztion, the guest VM's address space layout is locked and no more
```
Comment (atulkharerivos) [inline MEMORY.md:272]: s/finaliztion/finalization

File: MEMORY.md [hunk 2]
```diff
@@ -42,132 +65,279 @@ classDiagram
...
+a confidential region are considered to be owned by the VM and are guaranteed to
+be inaccessible to the VM's parent. Data pages, or pages initialized with
+host-provided contents, are measured as they are added and may only be added
+prior to VM finalization. Zero-filled confidential pages may be added at any
+point. Page faults in a confidential region trigger an exit to the VM's parent
+with the faulting address.
 
-# Memory assignment/sharing flows for TVMs and the hypervisor
+* `Shared`: Memory that is shared with the VM's parent. Pages in a shared
+memory region are considered to be owned by the parent VM, and the parent VM
```
Comment (atulkharerivos) [inline MEMORY.md:167]: s/"parent VM"/parent/g?

The parent may not necessarily be a VM.

File: MEMORY.md [hunk 3]
```diff
@@ -1,37 +1,60 @@
...
+time page allocator, `HypPageAlloc`, then uses `PageMap` to find and allocate
+free RAM pages for Salus-internal use. Once initialization has completed and all
+boot-time allocations have been made, the remaining free pages in `HypPageAlloc`
+are used to create `HostVmLoader`, which is to load and launch the host VM,
+and `PageTracker`, which uses `PageMap` to coordinate memory ownership
+and state transitions at runtime.
+
+`HostVmLoader` creates `HostVm`, which holds the Salus-internal state of the
+host VM. `HostVm` is itself composed of `VmPages`, which manages the guest
+physica address space of single VM. The remaining free pages are used as physical
```
Comment (atulkharerivos) [inline MEMORY.md:45]: s/physica/physical

**Review by atulkharerivos — APPROVED** [review]
Thanks for putting this together. It looks great, except for some minor nits. We will want to tweak it a little when we get around to the nested-TVM case (specifically, some references to host will have be reworded).

File: MEMORY.md (hunk 1 above)
Comment (abrestic-rivos) [reply MEMORY.md]: Done.

File: MEMORY.md (hunk 2 above)
Comment (abrestic-rivos) [reply MEMORY.md:167]: Hm? Every entity in the Salus model is a VM. (Well, except for the hypervisor itself, but it does not share memory and memory cannot be returned to it).

File: MEMORY.md (hunk 3 above)
Comment (abrestic-rivos) [reply MEMORY.md]: Done.

**Pull request merged**